### DISCUSSION TITLE
Why does a non-required prop does not accept `null`?

### DISCUSSION THREAD

**Discussion by LeBenLeBen** [vuejs/language-tools#2497] [Q&A]
I use a component from a library that has the following props definition:

```js
defineComponent({
  props: {
    modelValue: {
      type: [String, Number],
      default: null
    }
  }
}
```

When I pass `null` as a value for `modelValue`, it works fine with Vue (no props warning), but `vue-tsc` does not allow it and throws the following error:

```
(property) modelValue?: string | number | undefined
Type 'null' is not assignable to type 'string | number | undefined'.ts(2322)
```

I’m wondering why it allows `undefined` but not `null`? This is different from Vue internal props validator. Having to explicitly accept `null` for every optional props sounds weird to me, but I’m probably missing something?

**Comment by johnsoncodehk** [comment]
This is `defineComponent` type behavior, you can try `type: [String, Number] as PropType<string | number | null>` instead of.
